RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -151,120 +157,149 @@ namespace mkfit {
         y[ts] = tk.y();
         z[ts] = tk.z();
         d0[ts] = tk.d0BeamSpot(bspot.x, bspot.y);
+
+        b.register_entry_safe(
+            oldPhi[ts],
+            eta[ts]);  // If one is sure values are *within* axis ranges: b.register_entry(oldPhi[ts], eta[ts]);
       }
 
-      for (int ts = 0; ts < ns; ts++) {
+      b.finalize_registration();
+
+      for (int sorted_ts = 0; sorted_ts < ns; sorted_ts++) {
+        int ts = b.m_ranks[sorted_ts];
+
         if (not writetrack[ts])
-          continue;  // Note: this speed up prevents transitive masking (possibly marginal gain).
+          continue;  //FIXME: this speed up prevents transitive masking; check build cost!
 
         const float oldPhi1 = oldPhi[ts];
         const float pos2_first = pos2[ts];
-        const float eta1 = eta[ts];
-        const float pt1 = pt[ts];
+        const float Eta1 = eta[ts];
+        const float Pt1 = pt[ts];
         const float invptq_first = invptq[ts];
 
         // To study some more details -- need EventOfHits for this
         int n_ovlp_hits_added = 0;
+        // int  n_ovlp_hits_same_module = 0;
+        // int  n_ovlp_hits_shared = 0;
+        // int  n_ovlp_tracks = 0;
 
-        for (int tss = ts + 1; tss < ns; tss++) {
-          const float pt2 = pt[tss];
+        //these two loops may be put inside the class
+        auto phi_rng = ax_phi.Rrdr_to_N_bins(oldPhi[ts], 0.08);
+        auto eta_rng = ax_eta.Rrdr_to_N_bins(eta[ts], .1);
```

Comment:
```suggestion
        auto eta_rng = ax_eta.rrdr_to_N_bins(eta[ts], .1f);
```
- it looks like the capitalization updated did not get applied to the `binnor.h`. We'll likely be asked to fix.
- Here and in other places, I'd suggest to consistently use the the float literals  (.1f instead of .1) to avoid accidental type conversions

Reply:
I intentionally left this one as is ... R is a template parameter for the class. So this function name means: take r and delta-r values of the R type and convert them into bin-ranges in the N (broader) bin-space.

Probalbly it makes sense to be even more precise and use: convert_Rrange_from_rdr_to_N_bins()

Reply:
> Probalbly it makes sense to be even more precise and use: convert_Rrange_from_rdr_to_N_bins()

sounds reasonable